lib/attrsets.nix

```diff
@@ -248,11 +282,19 @@ rec {
     listToAttrs (map (n: nameValuePair n (f n)) names);
 
 
-  /* Check whether the argument is a derivation. */
-  isDerivation = x: isAttrs x && x ? type && x.type == "derivation";
+  /* Check whether the argument is a derivation. Any set with
+     { type = "derivation"; } counts as a derivation.
 
+     Example:
+       nixpkgs = import <nixpkgs> {}
+       isDerivation nixpkgs.ruby
+       => true
+       isDerivation "foobar"
+       => false
+  */
+  isDerivation = x: isAttrs x && x ? type && x.type == "derivation";
```

Comment:
Side note: Should just have `isDerivation = x: (x.type or null) == "derivation"`